applications/sintering/include/pf-applications/sintering/operator_advection.h

```diff
@@ -391,13 +393,25 @@ namespace Sintering
                           // Filter to detect grain boundary
                           auto etai_etaj = eta_i * eta_j;
 
-                          // Disable this normalization
+                          // Normalize or not
                           if (cgb >= 0)
-                            etai_etaj = compare_and_apply_mask<
-                              SIMDComparison::greater_than>(etai_etaj,
-                                                            cgb_lim,
-                                                            ones,
-                                                            zeros);
+                            {
+                              if (smoothening > 0)
+                                etai_etaj =
+                                  ones /
+                                  (ones + std::exp(-smoothening *
+                                                   (etai_etaj - cgb_lim)));
+                              else
+                                etai_etaj = compare_and_apply_mask<
+                                  SIMDComparison::greater_than>(etai_etaj,
+                                                                cgb_lim,
+                                                                ones,
+                                                                zeros);
+                            }
+                          else
+                            {
+                              etai_etaj *= 4.;
+                            }
```

Comment:
The else branch is new. Could you explain this?

Reply:
Yes, it is indeed new. This is done for the terms now to be always of the same orders of magnitude. The discontinuous or continuous filter vary within range [0, 1]. The multiplication of $\eta_i \eta_j$ - in range [0, 0.25]. So this is just to adjust the ranges. Of course, one can simply rescale the factor $k$, but that is less convenient and more prone to "oh I forgot" errors.